pkg/api/networkservice/mechanisms/vfio/helpers.go

```diff
@@ -17,51 +18,124 @@
 	if m.GetType() == MECHANISM {
-		return &mechanism{
+		return &Mechanism{
 			m,
 		}
 	}
 	return nil
 }
 
-// GetParameters returns map of mechanism parameters
-func (m *mechanism) GetParameters() map[string]string {
-	if m == nil {
-		return nil
+// GetParameters returns the map of all parameters to the mechanism
+func (m *Mechanism) GetParameters() map[string]string {
+	if m.Parameters == nil {
+		m.Parameters = map[string]string{}
 	}
 	return m.Parameters
 }
 
-// GetParameters returns network namespace Inode
-func (m *mechanism) GetNetNSInode() string {
-	if m == nil || m.GetParameters() == nil {
-		return ""
-	}
-	return m.GetParameters()[common.NetNSInodeKey]
+// GetCgroupDir returns client on host cgroup directory
+func (m *Mechanism) GetCgroupDir() string {
+	return m.GetParameters()[CgroupDirKey]
+}
+
+// SetCgroupDir sets client on host cgroup directory
+func (m *Mechanism) SetCgroupDir(cgroupDir string) {
+	m.Parameters[CgroupDirKey] = cgroupDir
+}
+
+// GetIommuGroup returns IOMMU group id
+func (m *Mechanism) GetIommuGroup() uint {
+	return atou(m.Parameters[IommuGroupKey])
+}
+
+// SetIommuGroup sets IOMMU group id
+func (m *Mechanism) SetIommuGroup(iommuGroup uint) {
+	m.Parameters[IommuGroupKey] = utoa(iommuGroup)
+}
+
+// GetPCIAddress returns PCI address
+func (m *Mechanism) GetPCIAddress() string {
+	return m.GetParameters()[PCIAddressKey]
+}
+
+// SetPCIAddress sets PCI address
+func (m *Mechanism) SetPCIAddress(pciAddress string) {
+	m.Parameters[PCIAddressKey] = pciAddress
+}
+
+// GetVfioMajor returns /dev/vfio major number
+func (m *Mechanism) GetVfioMajor() uint32 {
+	return uint32(atou(m.Parameters[VfioMajorKey]))
 }
 
-// GetPCIAddress returns PCI address of the VF device
-func (m *mechanism) GetPCIAddress() string {
-	if m == nil || m.GetParameters() == nil {
-		return ""
+// SetVfioMajor sets /dev/vfio major number
+func (m *Mechanism) SetVfioMajor(vfioMajor uint32) {
+	m.Parameters[VfioMajorKey] = utoa(uint(vfioMajor))
+}
+
+// GetVfioMinor returns /dev/vfio minor number
+func (m *Mechanism) GetVfioMinor() uint32 {
+	return uint32(atou(m.Parameters[VfioMinorKey]))
+}
+
+// SetVfioMinor sets /dev/vfio minor number
+func (m *Mechanism) SetVfioMinor(vfioMinor uint32) {
+	m.Parameters[VfioMinorKey] = utoa(uint(vfioMinor))
+}
+
+// GetDeviceMajor returns /dev/${iommuGroup} major number
+func (m *Mechanism) GetDeviceMajor() uint32 {
+	return uint32(atou(m.Parameters[DeviceMajorKey]))
+}
+
+// SetDeviceMajor sets /dev/${iommuGroup} major number
+func (m *Mechanism) SetDeviceMajor(deviceMajor uint32) {
+	m.Parameters[DeviceMajorKey] = utoa(uint(deviceMajor))
+}
+
+// GetDeviceMinor returns /dev/${iommuGroup} minor number
+func (m *Mechanism) GetDeviceMinor() uint32 {
+	return uint32(atou(m.Parameters[DeviceMinorKey]))
+}
+
+// SetDeviceMinor sets /dev/${iommuGroup} minor number
+func (m *Mechanism) SetDeviceMinor(deviceMinor uint32) {
+	m.Parameters[DeviceMinorKey] = utoa(uint(deviceMinor))
```

Comment:
Won't this panic if m == nil or Parameters == nil?

Reply:
Fixed, now GetParameters() returns empty map for m == nil.